src/cf_gen_defines

```diff
@@ -24,6 +24,8 @@ BEGIN {
 	define["FOLLOW_X_FORWARDED_FOR&&LINUX_NETFILTER"]="--enable-follow-x-forwarded-for and --enable-linux-netfilter"
 	define["FOLLOW_X_FORWARDED_FOR&&USE_ADAPTATION"]="--enable-follow-x-forwarded-for and (--enable-icap-client and/or --enable-ecap)"
 	define["FOLLOW_X_FORWARDED_FOR&&USE_DELAY_POOLS"]="--enable-follow-x-forwarded-for and --enable-delay-pools"
+	define["HAVE_AUTH_MODULE_BASIC"]="--enable-auth-basic"
+	define["HAVE_AUTH_MODULE_DIGEST"]="--enable-auth-digest"
```

Comment:
What effect, if any, does this correction have on various squid.conf files generated during Squid build?

Reply:
Both this and the `IF`/`ENDIF` tracking change the squid.conf.documented and HTML output to use the `./configure` option(s) listed in this array instead of the code internal `USE_*`/`HAVE_*` macro names.

For example, text at http://www.squid-cache.org/Doc/config/auth_param/
```
IF HAVE_AUTH_MODULE_BASIC
	=== Basic authentication parameters ===

...
		processing and similar.

ENDIF
IF HAVE_AUTH_MODULE_DIGEST
	=== Digest authentication parameters ===
...

ENDIF
```
becomes
```
if --enable-auth-basic
	=== Basic authentication parameters ===

...
		processing and similar.

endif
if --enable-auth-digest
	=== Digest authentication parameters ===
...

endif
```

Reply:
Great, thank you.